src/test/java/com/patreon/euphrates/ScrubbingInputStreamTest.java

```diff
@@ -0,0 +1,47 @@
+package com.patreon.euphrates;
+
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+import org.apache.commons.io.IOUtils;
+
+import java.io.ByteArrayInputStream;
+
+import static org.junit.Assert.*;
+
+public class ScrubbingInputStreamTest extends TestCase {
```

Comment:
Add some tests for reading data that's larger than the buffer, e.g. 5000 chars.  Make sure the buffer reset/flush/compact logic is right.